esx_service/vmdk_ops.py

```diff
@@ -702,6 +752,13 @@ def disk_attach(vmdk_path, vm):
     kv_status_attached, kv_uuid, attach_mode = getStatusAttached(vmdk_path)
     logging.info("Attaching {0} as {1}".format(vmdk_path, attach_mode))
 
+    # If the volume is attached then check if the attach is stale, detach
+    # the disk from the VM its attached to.
+    if kv_status_attached and kv_uuid != vm.config.uuid:
```

Comment:
I still think we should bother to look into KV Only if the attach failed. It makes logic a bit more complex though, so I am fine with the current approach if you feel the code after failure will be too convoluted. Comments?